File: src/main/scala/com/pwootage/oc/js/api/JSBiosInternalAPI.scala
package com.pwootage.oc.js.api

import com.pwootage.oc.js.jsvalue.JSValue
import com.pwootage.oc.js.{JSEngine, OCJS}
import li.cil.oc.api.machine.Machine

class JSBiosInternalAPI(machine: Machine, scriptEngine: JSEngine) {
  def log(msg: String): Unit = {
    // No-op for speeeed
    OCJS.log.error(msg)
  }

  def crash(msg: String): Unit = {
    machine.crash(msg)
    throw new JSExitException("JS Crashed: " + msg)
  }
}

File: src/main/java/com/pwootage/oc/js/api/JSBiosInternalAPI.kt
package com.pwootage.oc.js.api

import com.pwootage.oc.js.JSEngine
import com.pwootage.oc.js.OCJS

import li.cil.oc.api.machine.Machine

class JSBiosInternalAPI(val machine: Machine, val scriptEngine: JSEngine) {
  fun log(msg: String): Unit {
    // TODO: no-op this (or disable it)
    OCJS.log.error(msg)
  }

  fun crash(msg: String): Unit {
    machine.crash(msg)
    throw JSExitException("JS Crashed: $msg")
  }
}
